pkg/controllers/disruption/drift.go

```diff
@@ -51,11 +51,11 @@ func NewDrift(kubeClient client.Client, cluster *state.Cluster, provisioner *pro
 
 // ShouldDisrupt is a predicate used to filter candidates
 func (d *Drift) ShouldDisrupt(ctx context.Context, c *Candidate) bool {
-	return c.NodeClaim.StatusConditions().Get(string(d.Reason())).IsTrue()
+	return c.NodePool.Spec.Replicas == nil && c.NodeClaim.StatusConditions().Get(string(d.Reason())).IsTrue()
```

Comment:
I thought of a better way to add filtering during Candidate selection based on the disruption method, but that is going to grow the diff so I am keeping the filtering logic here